core/trino-main/src/main/java/io/trino/dispatcher/QueuedStatementResource.java

```diff
@@ -214,25 +214,21 @@ public void getStatus(
     {
         Query query = getQuery(queryId, slug, token);
 
-        // wait for query to be dispatched, up to the wait timeout
-        ListenableFuture<Void> futureStateChange = addTimeout(
-                query.waitForDispatched(),
-                () -> null,
-                WAIT_ORDERING.min(MAX_WAIT_TIME, maxWait),
-                timeoutExecutor);
-
-        // when state changes, fetch the next result
-        ListenableFuture<QueryResults> queryResultsFuture = Futures.transform(
-                futureStateChange,
-                ignored -> query.getQueryResults(token, uriInfo),
-                responseExecutor);
-
-        // transform to Response
-        ListenableFuture<Response> response = Futures.transform(
-                queryResultsFuture,
-                queryResults -> createQueryResultsResponse(queryResults, compressionEnabled),
-                directExecutor());
-        bindAsyncResponse(asyncResponse, response, responseExecutor);
+        ListenableFuture<Response> future = getStatus(query, token, maxWait, uriInfo);
+        bindAsyncResponse(asyncResponse, future, responseExecutor);
+    }
+
+    private ListenableFuture<Response> getStatus(Query query, long token, Duration maxWait, UriInfo uriInfo)
+    {
+        long waitMillis = WAIT_ORDERING.min(MAX_WAIT_TIME, maxWait).toMillis();
+
+        return FluentFuture.from(query.waitForDispatched())
```

Comment:
Code looks shorter. make sure the commit message includes rationale for the change, since with futures it's never obvious.

In particular, a person looking at the commit, will want to know:

- is this just an improvement of code readability?
- does this impact future cancellation? and cancellation propagation?

Reply:
Good point. I added a description that this is to improve readability but does not change behavior:

* `MoreFutures.addTimeout` uses `FluentFuture` in the same way
* `FluentFuture.transform` uses `Futures.transform`